jobs/scripts/sink-clustered-deployment/sink-clustered-deployment.sh

```diff
@@ -70,8 +70,8 @@ if [ -n "${ghprbPullId}" ]; then
 	CI_IMG_TAG="ci-k8s-${KUBE_VERSION}-pr${ghprbPullId}"
 	# if the sha1 hash is provided, we will try to append a short form of it to
 	# the tag to make the image unique to each "push" of the PR.
-	shortsha="${sha1:0:8}"
-	if [ -n "${shortsha}" ]; then
+	if [[ "$sha1" =~ ^[abcdef0-9]{4}[abcdef0-9]*$ ]]; then
```

Comment:
Why `{4}` and not `{8}`? Above regex would match a hash with just 4 characters from the list.

Reply:
> The intent was the allow the sha1 variable to be as short as 4 chars (first part of the regex), but only allow valid hex chars (second part of the regex). The $ at the end anchors the match so that nothing "extra" can follow the char sets of [abcdef0-9] -- at least that's the intent. So it would accept abcd or abcd1234, etc, but not something like origin/master. I thought four was sufficient for uniqueness. 

I agree. The next line basically extracts first 8 characters which doesn't match the regex here. That's the reason I asked about `{4}`.

Reply:
Right, it prefers 8 but will accept as few as 4.  After all:
```
$ foo=asdf; echo "${foo:0:8}"
asdf
$ foo=asdfjklqwertyuiop; echo "${foo:0:8}"
asdfjklq
```